events/code_review_events/workflow.py

```diff
@@ -216,78 +252,145 @@ def publish_results(self, payload):
+                t
+                for t in selected_tasks["response"].text.splitlines()
+                if t.startswith("test-linux64/")
+            ]
+        }
+        if len(selected_tasks["tasks"]) == 0:
+            return None
 
-                # Send task to monitoring
-                await self.bus.send(
-                    QUEUE_MONITORING,
-                    ("project-relman", "bugbug-classify-patch", task_id),
-                )
-        except Exception as e:
-            logger.error("Failed to trigger risk analysis task", error=str(e))
+        # Get the decision task of the push to try.
+        decision_task_index = self.index_service.findTask(
+            "gecko.v2.try.revision.{}.taskgraph.decision".format(revision)
+        )
+        decision_task_id = decision_task_index["taskId"]
 
-    def should_run_risk_analysis(self, build):
-        """
-        Check if we should trigger a risk analysis for this revision:
-        * when the revision is being reviewed by one of some specific reviewers
-        """
-        if self.community_hooks is None:
-            return False
+        # Find the 'add-new-jobs' action to add new jobs to the task group.
+        actions = self.queue_service.getLatestArtifact(
+            decision_task_id, "public/actions.json"
+        )
+        add_job_action = next(
+            action for action in actions["actions"] if action["name"] == "add-new-jobs"
+        )
+        assert add_job_action["kind"] == "hook"
+
+        # Trigger the 'add-new-jobs' action with the list of tasks bugbug selected.
+        jsonschema.validate(instance=selected_tasks, schema=add_job_action["schema"])
+
+        hookPayload = jsone.render(
+            add_job_action["hookPayload"],
+            context={
+                "taskId": None,
+                "taskGroupId": decision_task_id,
+                "input": selected_tasks,
+            },
+        )
 
-        usernames = set(
-            [reviewer["fields"]["username"] for reviewer in build.reviewers]
+        add_job_task = self.hooks_service.triggerHook(
+            add_job_action["hookGroupId"], add_job_action["hookId"], hookPayload
+        )
+        add_job_task_id = add_job_task["status"]["taskId"]
+        logger.info(
+            "Triggered a add-new-jobs task to add a set of test tasks",
+            task_group=decision_task_id,
+            id=add_job_task_id,
         )
-        return len(usernames.intersection(self.risk_analysis_reviewers)) > 0
 
-    def should_run_test_selection(self, build):
-        """
-        Check if we should trigger a test selection for this revision:
-        * randomly for a subset of revisions
-        """
-        if self.community_hooks is None:
-            return False
+        return decision_task_id
 
-        return random.random() < taskcluster_config.secrets.get(
-            "test_selection_share", 0.0
-        )
+    async def got_bugbug_test_select_end(self, payload):
```

Comment:
Why not put that in bugbug_utils too ? You could store the `diff_to_push` there or on a separate singleton to pave the way for persistence